### DISCUSSION TITLE
LFS URL when using git sparse clone that does not include .lfsconfig

### DISCUSSION THREAD

**Discussion by lucasrangit** [git-lfs/git-lfs#4991] [Q&A]
If I perform a git sparse clone and include paths that have LFS objects but don't include the `.lfsconfig` how does `git-lfs` know where to pull the LFS objects from?

Here's the flow:
 > git init repo
 > git fetch --no-tags --force --progress --depth=1 -- git@repo.git +refs/heads/master:refs/remotes/origin/master
 > git config remote.origin.url git@repo.git
 > git config --add remote.origin.fetch +refs/heads/master:refs/remotes/origin/master 
 > git config core.sparsecheckout
 > git config lfs.fetchinclude .gitattributes,.gitignore
 > git checkout -f commit_sha
 > git config --get remote.origin.url
 > git lfs pull origin

This is working but I'm not sure which URL is being used. I can't easily instrument because it's a remote build system. Which is why I am asking here.

Thank you.

**Comment by bk2204** [comment]
Hey,

First, Git LFS isn't currently tested with sparse checkout, so it may or may not work.  However, having said that, the documentation in `git-lfs-config(5)` indicates the ordering here:

> If the `.lfsconfig` file is missing, the index is checked for a version of the file, and that is used instead.  If both are missing, `HEAD` is checked for the file.  If the repository is bare, only `HEAD` is checked.  This order may change for checkouts in the future to better match Git's behavior.
